### PR TITLE
Re-format and lint code using black and ruff

### PR DESCRIPTION
**Pull request by csris**
Run the black formatter from PSF to standardize code style; run ruff linter and fix all errors to address anti-patterns.

### CODE REVIEW

**Comment by csris** [comment]
@jrb, can you take a look, especially at the workflows? Not sure if this is a security risk.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[tool.black]
+target-version = ['py310']
```
Comment (orangetin) [inline pyproject.toml:2]: the docker image has python=3.8.10 installed. Not a big deal.

The rest looks good to me.

Comment (csris) [reply pyproject.toml:2]: We should probably update the Docker image soon.

**Comment by csris** [comment]
Updated.

File: pyproject.toml (hunk 1 above)
Comment (orangetin) [reply pyproject.toml]: > We should probably update the Docker image soon.

Made a ticket for this

**Comment by csris** [comment]
Closing; no longer needed.

**Pull request closed without merging**